ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -147,50 +147,112 @@ def get_partial_event_contingency_tables(self) -> List[EventContingencyTable]:
         """
         funnel_persons_query, funnel_persons_params = self.get_funnel_persons_cte()
 
-        # TODO: replace countIf with the distinct equivalent
+        non_deleted_distinct_ids = f"""
+            SELECT distinct_id,
+                argMax(person_id, _timestamp) as person_id
+            FROM (
+                SELECT distinct_id,
+                        person_id,
+                        max(_timestamp) as _timestamp
+                    FROM person_distinct_id
+                    WHERE team_id = {self._team.pk}
+                    GROUP BY person_id,
+                            distinct_id,
+                            team_id
+                HAVING max(is_deleted) = 0
+            )
+            GROUP BY distinct_id
+        """
+
         query = f"""
             WITH 
-                funnel_people AS ({funnel_persons_query}),
                 toDateTime(%(date_to)s) AS date_to,
-                %(target_step)s AS target_step
+                toDateTime(%(date_from)s) AS date_from,
+                %(target_step)s AS target_step,
+                %(funnel_step_names)s as funnel_step_names
+
             SELECT 
                 event.event AS name, 
 
-                -- If we have a timestamp, we know the person reached the end of the
-                -- funnel (I think)
-                countIf(person.steps = target_step) AS success_count,
+                -- If we have a `person.steps = target_step`, we know the person 
+                -- reached the end of the funnel
+                countDistinctIf(
+                    person.person_id, 
+                    person.steps = target_step
+                ) AS success_count,
 
                 -- And the converse being for failures
-                countIf(person.steps <> target_step) AS failure_count
+                countDistinctIf(
+                    person.person_id, 
+                    person.steps <> target_step
+                ) AS failure_count
+
             FROM events AS event
             
-            JOIN person_distinct_id AS pdi
-                ON pdi.distinct_id = events.distinct_id
-
-            -- Right join, so we can get the total success/failure numbers as well
-            RIGHT JOIN funnel_people AS person
+            JOIN ({non_deleted_distinct_ids}) AS pdi
+                ON pdi.distinct_id = events.distinct_id AND pdi.team_id = event.team_id
```

Comment:
The current test errors should go away after:
```suggestion
                ON pdi.distinct_id = events.distinct_id
```